packages/extra/src/OffsetDate.js

```diff
@@ -0,0 +1,1274 @@
+        const epochSec = instant.epochSecond() + offset.totalSeconds();  // overflow caught later
+        const epochDay = MathUtil.floorDiv(epochSec, SECONDS_PER_DAY);
+        const date = LocalDate.ofEpochDay(epochDay);
+        return new OffsetDate(date, offset);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Obtains an instance of `OffsetDate` from a temporal object.
+     * 
+     * A `TemporalAccessor` represents some form of date and time information.
+     * This factory converts the arbitrary temporal object to an instance of `OffsetDate`.
+     * 
+     * The conversion extracts and combines `LocalDate` and `ZoneOffset`.
+     * 
+     * This method matches the signature of the functional interface {@link TemporalQuery}
+     * allowing it to be used in queries via method reference, `OffsetDate.FROM`.
+     *
+     * @param {TemporalAccessor} temporal  the temporal object to convert, not null
+     * @return {OffsetDate} the offset date, not null
+     * @throws DateTimeException if unable to convert to an `OffsetDate`
+     */
+    static from(temporal) {
+        if (temporal instanceof OffsetDate) {
+            return temporal;
+        }
+        requireNonNull(temporal, 'temporal');
+        try {
+            const date = LocalDate.from(temporal);
+            const offset = ZoneOffset.from(temporal);
+            return new OffsetDate(date, offset);
+        } catch (ex) {
+            throw new DateTimeException(`Unable to obtain OffsetDate from TemporalAccessor: ${temporal.constructor.name}`, ex);
+        }
+    }
+
+    /**
+     * Obtains an instance of `OffsetDate` from a text string using a specific formatter.
+     * 
+     * The text is parsed using the formatter, returning a date.
+     *
+     * @param {CharSequence} text  the text to parse, not null
+     * @param {DateTimeFormatter} formatter  the formatter to use, not null
+     * @return {OffsetDate} the parsed offset date, not null
+     * @throws DateTimeParseException if the text cannot be parsed
+     */
+    static parse(text, formatter = DateTimeFormatter.ISO_OFFSET_DATE) {
+        requireNonNull(formatter, 'formatter');
+        return formatter.parse(text, OffsetDate.FROM);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Constructor.
+     *
+     * @param {LocalDate} date  the local date, not null
+     * @param {ZoneOffset} offset  the zone offset, not null
+     */
+    constructor(date, offset) {
+        super();
+        this._date = requireNonNull(date, 'date');
+        this._offset = requireNonNull(offset, 'offset');
+    }
+
+    /**
+     * Returns a new date based on this one, returning `this` where possible.
+     *
+     * @param {LocalDate} date  the date to create with, not null
+     * @param {ZoneOffset} offset  the zone offset to create with, not null
+     */
+    _with(date, offset) {
```

Comment:
Why to we have a `_with` but no implementation for the overide of `with(...)` ? Do i oversee something ?

Reply:
`OffsetDate._with` is not related to `Temporal.with`. See https://github.com/ThreeTen/threeten-extra/blob/v1.7.1/src/main/java/org/threeten/extra/OffsetDate.java#L327.

Underscore needs to be here because 1. this method is not public, 2. even if not, it cant be named `with` because it would overload `Temporal.with`.

Same goes for `YearQuarter._with`.